src/roman_datamodels/stnode/_node.py

```diff
@@ -38,22 +38,20 @@ def __init__(self, node=None, parent=None, name=None):
         self._parent = parent
         self._name = name
 
-    @staticmethod
-    def _convert_to_scalar(key, value, ref=None):
-        """Find and wrap scalars in the appropriate class, if its a tagged one."""
-        from ._tagged import TaggedScalarNode
+    def _wrap_value(self, key, value):
+        # Return objects as node classes, if applicable
+        if isinstance(value, dict | AsdfDictNode):
+            return DNode(value, parent=self, name=key)
 
-        if isinstance(ref, TaggedScalarNode):
-            # we want the exact class (not possible subclasses)
-            if type(value) == type(ref):  # noqa: E721
-                return value
-            return type(ref)(value)
+        elif isinstance(value, list | AsdfListNode):
+            return LNode(value)
 
-        if isinstance(value, TaggedScalarNode):
+        else:
             return value
 
-        if key in SCALAR_NODE_CLASSES_BY_KEY:
-            value = SCALAR_NODE_CLASSES_BY_KEY[key](value)
+    @staticmethod
+    def _convert_to_scalar(key, value, ref=None):
+        """For now this is a simple pass through so a mixin can override it."""
 
         return value
```

Comment:
We may want to indicate off the `_read_tag` so that old files still function the same.